website/server/controllers/api-v3/auth.js

```diff
@@ -121,10 +122,10 @@ api.registerLocal = {
       },
     };
 
-    if (fbUser) {
-      if (!fbUser.auth.facebook.id) throw new NotAuthorized(res.t('onlySocialAttachLocal'));
-      fbUser.auth.local = newUser.auth.local;
-      newUser = fbUser;
+    if (existingUser) {
+      if (!existingUser.auth.facebook.id && !existingUser.auth.google.id) throw new NotAuthorized(res.t('onlySocialAttachLocal'));
```

Comment:
Suggestion: loop through SUPPORTED_SOCIAL_NETWORKS instead of manually referencing facebook and google.